### PR TITLE
Fix Adapter\ReflectionClass::$name

### PR DESCRIPTION
**Pull request by ondrejmirtes**
Part of #673.

### CODE REVIEW

File: src/Reflection/Adapter/ReflectionClass.php [hunk 1]
```diff
@@ -68,6 +70,22 @@ public function __toString()
...
+     *
+     * @phpcsSuppress SlevomatCodingStandard.TypeHints.ParameterTypeHint.MissingNativeTypeHint
+     */
+    public function __get($name)
+    {
+        if ($name === 'name') {
+            return $this->betterReflectionClass->getName();
+        }
+
+        return $this->{$name};
```
Comment (Ocramius) [inline src/Reflection/Adapter/ReflectionClass.php:86]: I think we need to prevent this - makes properties public-ish by default

Comment (ondrejmirtes) [reply src/Reflection/Adapter/ReflectionClass.php:86]: I haven't realized this. As there are no other public properties we probably don't need to care about that? But I'm not sure how to trigger the same error PHP itself does when stumbling upon an undefined property...

Comment (Ocramius) [reply src/Reflection/Adapter/ReflectionClass.php:86]: `$this->betterReflectionClass`

Comment (ondrejmirtes) [reply src/Reflection/Adapter/ReflectionClass.php]: Just updated it.

File: src/Reflection/Adapter/ReflectionClass.php [hunk 2]
```diff
@@ -68,6 +70,22 @@ public function __toString()
...
+     *
+     * @phpcsSuppress SlevomatCodingStandard.TypeHints.ParameterTypeHint.MissingNativeTypeHint
+     */
+    public function __get($name)
+    {
+        if ($name === 'name') {
+            return $this->betterReflectionClass->getName();
+        }
+
+        return $this->betterReflectionClass->{$name};
```
Comment (Ocramius) [inline src/Reflection/Adapter/ReflectionClass.php:86]: Can we test this path, or maybe throw an exception?

Comment (Ocramius) [reply src/Reflection/Adapter/ReflectionClass.php:86]: Yep, throwing an exception seems more sensible here: we just need a test case for it :)

Comment (ondrejmirtes) [reply src/Reflection/Adapter/ReflectionClass.php:86]: I'd be more comfortable with 1:1 behaviour with PHP's reflection class :)

Comment (Ocramius) [reply src/Reflection/Adapter/ReflectionClass.php:86]: In which case you'd need a nightmare like https://github.com/Ocramius/ProxyManager/blob/master/src/ProxyManager/ProxyGenerator/Util/PublicScopeSimulator.php :P 

Ok, we just need a test raising an error then (and asserting on it)

Comment (ondrejmirtes) [reply src/Reflection/Adapter/ReflectionClass.php:86]: Oh god. Alright, what should be thrown?

Comment (Ocramius) [reply src/Reflection/Adapter/ReflectionClass.php:86]: an `OutOfBoundsException`, probably

Comment (ondrejmirtes) [reply src/Reflection/Adapter/ReflectionClass.php]: Updated.

**Pull request merged**